web_generator/lib/src/dart_main.dart

```diff
@@ -24,16 +26,44 @@ void main(List<String> args) async {
   if (languageVersionString.isEmpty) {
     languageVersionString = DartFormatter.latestLanguageVersion.toString();
   }
-  final ArgResults argResult;
-  argResult = _parser.parse(args);
-  await _generateAndWriteBindings(
-    outputDirectory: argResult['output-directory'] as String,
-    generateAll: argResult['generate-all'] as bool,
-    languageVersion: Version.parse(languageVersionString),
-  );
+
+  final argResult = _parser.parse(args);
+
+  switch (argResult.rest.first.toLowerCase()) {
+    case 'idl':
+      await generateIDLBindings(
+        outputDirectory: argResult['output'] as String,
+        generateAll: argResult['generate-all'] as bool,
+        languageVersion: Version.parse(languageVersionString),
```

Comment:
We'll probably need this still to format the generated files, but totally okay waiting to add this until then too.